resources/documentation/multiple_icebergs.rst

```diff
@@ -227,45 +291,226 @@ And add management2 to iceberg2, by editing /etc/bluebanquise/inventory/cluster/
+
+The Slurm setup is a little bit more complex. You will need a unified network to
+allow your nodes to be able to reach the same Slurm controller, mostly running
+on the mngt1 server.
+
+But you will also need to ensure direct hostnames resolution of all computes
+nodes is done on the interconnect, and not on the internet. Why ? Simply because
+when parallel computations take places, Slurm will provide to the instance nodes
+hostnames as target, and so if nodes need to reach each other through ethernet,
+nodes from once iceberg will not be able to reach nodes from other icebergs, and
+so parallel computations will not initialize.
+
+Example: user job is asking for 100 nodes, the whole cluster is free, and each
+iceberg contains 80 nodes. Slurm will allocate 80 nodes from iceberg1, lets say
+c[001-080] and 20 nodes from iceberg2, c[081-100]. The final mpirun command
+will receive as hosts target c[001-100]. c001 will be able to communicate with
+c002, but not with c081, as iceberg ethernet networks are isolated. But if c081
+means c081 over the interconnect network, then since this network is unified,
+c001 will be able to reach c081 and initialize MPI run.
+
+.. note::
+  It is possible to set routing between icebergs over ethernet, but this is not
+  in the scope of this documentation.
+
+To achieve direct computes hosts resolution over interconnect, ensure the
+interconnect network interface is first in the network_interface list of each
+compute node.
+
+For example:
 
 .. code-block:: text
 
-  ansible-playbook /etc/bluebanquise/playbooks/management2.yml
+  hosts:
+    c001:
+      bmc:
+        name: bc001
+        ip4: 10.2.103.1
+        mac: 08:00:27:0d:f8:a5
+        network: ice2-1
+      network_interfaces:
+        - interface: enp0s3
+          ip4: 10.2.3.1
+          mac: 08:00:27:0d:f8:a6
+          network: ice2-1
+        - interface: ib0
+          ip4: 10.20.3.1
+          network: interconnect-1
+
+Becomes:
+
+.. code-block:: text
 
-And proceed as usual to add more hosts into iceberg2 and deploy them, this time from management2.
+  hosts:
+    c001:
+      bmc:
+        name: bc001
+        ip4: 10.2.103.1
+        mac: 08:00:27:0d:f8:a5
+        network: ice2-1
+      network_interfaces:
+        - interface: ib0
+          ip4: 10.20.3.1
+          network: interconnect-1
+        - interface: enp0s3
+          ip4: 10.2.3.1
+          mac: 08:00:27:0d:f8:a6
+          network: ice2-1
+
+Using this, all nodes will now be able to communicate directly over the
+interconnect.
+
+.. note::
+  Having interconnect here as direct resolution is not an issue to deploy
+  configuration with Ansible. By default, the ss_master role force the ssh from
+  a management to targets to be done on the first management network in the
+  target network_interfaces list. In this example, a ping c001 will ping the
+  ib0 interface, so 10.20.3.1, but an ssh c001 will connect to c001 through
+  interface enp0s3, so 10.2.3.1 .
```

Comment:
through interface connected to the *iceX* network?

Reply:
Is it ok this way ?
  Having interconnect here as direct resolution is not an issue to deploy
  configuration with Ansible. By default, the ss_master role force the ssh from
  a management to targets to be done on the first management network in the
  target network_interfaces list. In this example, a ping c001 will ping the
  ib0 interface connected to the ib0 network, so 10.20.3.1, but an ssh c001 will
  connect to c001 through interface enp0s3 connected to the ice2-1 network, so
  10.2.3.1.

Reply:
yes